app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java

```diff
@@ -51,36 +68,164 @@ default MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSou
+                            throw new IOException("Error when generating the DASH manifest of "
+                                    + "YouTube OTF stream " + stream.getContent(), e);
+                        }
+                    } else if (streamType == StreamType.POST_LIVE_STREAM) {
+                        // If the content comes from YouTube, is not an URL, uses the DASH delivery
+                        // method and if the stream type of the stream is a post live stream, it
+                        // means that the content is an OTF stream so we need to generate the
+                        // manifest corresponding to the content (which is the last segment of the
+                        // stream)
+
+                        try {
+                            final ItagItem itagItem = Objects.requireNonNull(stream.getItagItem());
+                            content = YoutubeDashManifestCreator
+                                    .createDashManifestFromPostLiveStreamDvrStreamingUrl(
+                                            stream.getContent(), itagItem, itagItem
+                                                    .getTargetDurationSec());
+                        } catch (final YoutubeDashManifestCreator.
+                                YoutubeDashManifestCreationException | NullPointerException e) {
+                            throw new IOException("Error when generating the DASH manifest of "
+                                    + "YouTube ended live stream " + stream.getContent(), e);
+                        }
+                    } else {
+                        throw new IllegalArgumentException("DASH manifest generation of YouTube"
+                                + "livestreams is not yet supported");
+                    }
+                } else {
+                    // DASH manual streams from other stream services do not require manifest
+                    // generation
+                    content = stream.getContent();
+                }
+                final DashManifest dashManifest;
+                try {
+                    final ByteArrayInputStream dashManifestInput = new ByteArrayInputStream(
+                            content.getBytes(StandardCharsets.UTF_8));
+                    String baseUrl = stream.getBaseUrl();
+                    if (baseUrl == null) {
+                        baseUrl = "";
+                    }
+
+                    dashManifest = new DashManifestParser().parse(Uri.parse(baseUrl),
+                            dashManifestInput);
+                } catch (final IOException e) {
+                    throw new IOException("Error when parsing manual DASH manifest", e);
+                }
+
+                return dataSource.getDashMediaSourceFactory().setTag(metadata)
+                        .createMediaSource(dashManifest);
+            }
+        } else {
+            throw new IllegalArgumentException("Unsupported delivery type" + deliveryMethod);
+        }
+    }
```

Comment:
This method is long and deeply nested. Please split it up into multiple methods.

Reply:
I will split this method into three methods: one for DASH, one for progressive HTTP and one for HLS.

Reply:
The `buildMediaSource` method is now split into three methods, and the DASH method itself is split into three methods.